node/src/db/db_handlers.rs

```diff
@@ -0,0 +1,693 @@
+                (*parent_bead as u64),
+                (*bead_id as u64),
+                current_parent_timestamp.to_u32().to_u64().unwrap(),
+            ));
+        }
+        for bead_tx in bead.committed_metadata.transactions.iter() {
+            transaction_tuples.push(((*bead_id as u64), bead_tx.compute_txid().to_string()));
+        }
+        //Constructing json bindings
+        let transactions_values = transaction_tuples
+            .iter()
+            .map(|t| {
+                json!({
+                    "txid":t.1,
+                    "bead_id":t.0
+                })
+            })
+            .collect::<Vec<_>>();
+        let parent_timestamps_values = parent_timestamp_tuples
+            .iter()
+            .map(|p| {
+                json!({
+                    "child":p.1,
+                    "parent":p.0,
+                    "timestamp":p.2
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let relatives_values = relative_tuples
+            .iter()
+            .map(|r| {
+                json!({
+                    "parent":r.0,
+                    "child":r.1
+                })
+            })
+            .collect::<Vec<_>>();
+        let txs_json = serde_json::to_string(&transactions_values).unwrap();
+        let relative_json = serde_json::to_string(&relatives_values).unwrap();
+        let parent_timestamp_json = serde_json::to_string(&parent_timestamps_values).unwrap();
+        let hex_converted_version =
+            hex::encode(bead.block_header.version.to_consensus().to_be_bytes());
+        let hex_converted_nonce = hex::encode(bead.block_header.nonce.to_be_bytes());
+        let hex_converted_ntime = hex::encode(bead.block_header.time.to_u32().to_be_bytes());
+        #[allow(unused)]
+        let hex_converted_extranonce =
+            hex::encode(bead.uncommitted_metadata.extra_nonce.to_be_bytes());
+        //All fields are in be format
+        if let Err(e) = sqlx::query(&INSERT_QUERY)
+            .bind(*bead_id as i64)
+            .bind(bead.block_header.block_hash().to_string())
+            .bind(hex_converted_version)
+            .bind(bead.block_header.prev_blockhash.to_string())
+            .bind(bead.block_header.merkle_root.to_string())
+            .bind(hex_converted_ntime)
+            .bind(bead.block_header.bits.to_hex())
+            .bind(hex_converted_nonce)
+            .bind(bead.committed_metadata.payout_address)
+            .bind(bead.committed_metadata.start_timestamp.to_string())
+            .bind(bead.committed_metadata.comm_pub_key.to_string())
+            .bind(bead.committed_metadata.min_target.to_hex())
+            .bind(bead.committed_metadata.weak_target.to_hex())
+            .bind(bead.committed_metadata.miner_ip)
+            //TODO: Placeholder till splitting of `uncomitted_metadata` extranonce to extranonce1 + extranonce2
+            .bind("0101010101010101".to_string())
```

Comment:
This number is larger than i32::MAX so would cause a ParseIntError if ever saved to the db. Please make it 0.